template/web/wp-content/themes/<%= slug %>/app/Managers/CustomPostTypesManager.php

```diff
@@ -16,15 +17,52 @@ class CustomPostTypesManager implements ManagerInterface {
 	 */
 	public function run() {
 		add_action( 'init', array( $this, 'register_custom_post_types' ), 1 );
+		add_filter( 'Timber\PostClassMap', array( $this, 'set_timber_classmap' ) );
 	}
 
 	/**
 	 * Register custom post types.
 	 *
-	 * @todo use Studiometa\WP_Factory
+	 * Use PostTypeBuilder from studiometa/wp-toolkit.
+	 *
+	 * @see https://github.com/studiometa/wp-toolkit
+	 *
 	 * @return void
 	 */
-	public function register_custom_post_types() {
+	public function register_custom_post_types() {}
```

Comment:
**question:** Should we rename this method to `register_custom_post_type()` to indicate that CPT should be registered in their own method?

Reply:
I'm not sure to understand, do you want to have one method for each post type ? like `register_post_type_product()`, `register_post_type_testimonies()` ?

Reply:
Yes, that's the idea. I find the file easier to read and to manage to have one method per CPT.

Reply:
Should we have both ? To avoid adding multiple `add_action()`. 

Something like this :

```
class CustomPostTypesManager implements ManagerInterface {
	public function run() {
		add_action( 'init', array( $this, 'register_custom_post_types' ), 1 );
        }
       
      	public function register_custom_post_types() {
                 $this->register_post_type_product();
                 // Register other CPT bellow
        }

       public function register_post_type_product() {
                 // Register product CPT
       }
```

Reply:
What is the impact on performance of multiple `add_action()` vs one with multiple method calls? 

I find the pattern with multiple `add_action()` less complex as it has one less level:

![mermaid-diagram-20211220112523](https://user-images.githubusercontent.com/250145/146752534-146738cd-f209-41bd-b97f-4f79a6abdc8e.png)

I think we should use the pattern you suggest only if there is a meaningful impact on performance with the usage of multiple `add_action`.